cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/RightsizeRequest.java

```diff
@@ -42,26 +43,63 @@ protected RightsizeResult handle() {
     Provisioner provisioner = config.getConfiguredInstance(AnomalyDetectorConfig.PROVISIONER_CLASS_CONFIG,
                                                            Provisioner.class,
                                                            overrideConfigs);
-    Supplier<Set<String>> topicNameSupplier = () -> _kafkaCruiseControl.kafkaCluster().topics();
-    Set<String> topicNamesMatchedWithPattern = Utils.getTopicNamesMatchedWithPattern(_parameters.topic(), topicNameSupplier);
-    String topicName;
-    if (topicNamesMatchedWithPattern.size() > 1) {
-      throw new IllegalArgumentException(String.format("The RightsizeEndpoint does not support provisioning for multiple topics {%s}.",
-                                                       String.join(" ,", topicNamesMatchedWithPattern)));
-    } else {
-      topicName = topicNamesMatchedWithPattern.iterator().next();
-    }
-    ProvisionRecommendation recommendation =
-        new ProvisionRecommendation.Builder(ProvisionStatus.UNDER_PROVISIONED).numBrokers(_parameters.numBrokersToAdd())
-                                                                              .numPartitions(_parameters.partitionCount())
-                                                                              .topic(topicName)
-                                                                              .build();
+    ProvisionRecommendation recommendation = sanityCheckResources();
     Map<String, ProvisionRecommendation> provisionRecommendation;
     provisionRecommendation = Collections.singletonMap(RECOMMENDER_UP, recommendation);
 
     ProvisionerState provisionerState = provisioner.rightsize(provisionRecommendation, new RightsizeOptions());
 
-    return new RightsizeResult(_parameters.numBrokersToAdd(), _parameters.partitionCount(), topicName, provisionerState, config);
+    return new RightsizeResult(_parameters.numBrokersToAdd(), _parameters.partitionCount(), _topicName, provisionerState, config);
+  }
+
+  /**
+   * Ensure that
+   * <ul>
+   *   <li>exactly one resource type is set</li>
+   *   <li>if the resource type is partition, then the corresponding topic must be specified; otherwise, the topic cannot be specified</li>
+   *   <li>if the resource type is partition, then only one topic must be specified</li>
+   * </ul>
+   *
+   * @return The {@link ProvisionRecommendation} to recommend for rightsizing
+   * @throws IllegalArgumentException when the sanity check fails.
+   */
+  private ProvisionRecommendation sanityCheckResources() throws IllegalArgumentException {
```

Comment:
The name "sanity check" typically indicates a function that simply verifies whether some required conditions are satisfied -- i.e. it does not modify the state. 
Should we rename this function to something that better describes its functionality -- e.g. `createProvisionRecommendation`?